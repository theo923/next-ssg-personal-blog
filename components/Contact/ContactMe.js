import React from "react";

const initialState = {
    title: "",
    email: "",
    comment: "",
    complete: false,
};

class ContactMe extends React.Component {
    constructor() {
        super();
        this.state = initialState;
    }

    async insertData(commentdata) {
        const response = await fetch("/api/contact", {
            method: "POST",
            headers: {
                "Content-Type": "application/json",
                Accept: "application/json",
            },
            body: JSON.stringify({
                data: commentdata,
            }),
        });
    }

    handleChange = (e) => {
        const { name, value } = e.target;
        this.setState({ [name]: value });
    };

    reset = () => {
        this.setState(initialState);
    };

    render() {
        const { title, email, comment, complete } = this.state;
        return (
            <section data-test="component-contact">
                <div className="container contactWindow">
                    <div className="contact">
                        {!complete ? (
                            <div className="flex flex-col">
                                <div className="contactHeading">Comment</div>
                                <div className="contactGrid">
                                    <label htmlFor="title">Title:</label>
                                    <input
                                        name="title"
                                        value={title}
                                        onChange={(e) => this.handleChange(e)}
                                    ></input>
                                    <label htmlFor="email">
                                        Email Address:
                                    </label>
                                    <input
                                        name="email"
                                        value={email}
                                        onChange={(e) => this.handleChange(e)}
                                    ></input>
                                    <label htmlFor="comment">Comment:</label>
                                    <textarea
                                        name="comment"
                                        value={comment}
                                        onChange={(e) => this.handleChange(e)}
                                    ></textarea>
                                    <div className="hidden md:block"></div>
                                    <div className="hidden md:block"></div>
                                    <div className="contactActionBtn">
                                        <button
                                            type="submit"
                                            onClick={() => {
                                                this.insertData({
                                                    title,
                                                    email,
                                                    comment,
                                                });
                                                this.setState((prevState) => ({
                                                    complete:
                                                        !prevState.complete,
                                                }));
                                            }}
                                        >
                                            Submit
                                        </button>
                                        <button onClick={() => this.reset()}>
                                            Reset
                                        </button>
                                    </div>
                                </div>
                            </div>
                        ) : (
                            <div className="font-juliusSansOne text-3xl">
                                Thanks for your comment !
                            </div>
                        )}
                    </div>
                </div>
            </section>
        );
    }
}

export default ContactMe;
