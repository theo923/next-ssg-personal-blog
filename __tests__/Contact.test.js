import { shallow } from 'enzyme';
import React from 'react';
import ContactMe from '../components/Contact/ContactMe';

describe('ContactMe test', () => {

    it("expect ContactMe component is rendered without crashing", () => {
        const wrapper = shallow(<ContactMe />);   
    });

    it("expect ContactMe component is rendered && match with the Snapshot", () => {
        const wrapper = shallow(<ContactMe />);   
        expect(shallow(<ContactMe />)).toMatchSnapshot();
    });
    
});





