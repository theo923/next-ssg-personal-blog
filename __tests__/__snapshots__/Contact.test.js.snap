// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactMe test expect ContactMe component is rendered && match with the Snapshot 1`] = `
<div>
  <section
    className="text-blueGray-700"
  >
    <div
      className="container items-center px-5 py-64 mx-auto h-full lg:px-16"
    >
      <div
        className="mb-12 text-left"
      >
        <div
          className="p-6 my-4 lg:p-8 rounded-xl bg-gray-100 shadow-2xl"
        >
          <div
            className="flex flex-col"
          >
            <div
              className="font-juliusSansOne font-bold text-4xl self-center py-3 mb-4"
            >
              Comment
            </div>
            <div
              className="grid grid-cols-1 md:grid-cols-3"
            >
              <label
                className="font-juliusSansOne text-xl py-2"
                htmlFor="title"
              >
                Title:
              </label>
              <input
                className="pr-20 py-2 md:pr-40 md:my-2 md:col-span-2"
                name="title"
                onChange={[Function]}
                value=""
              />
              <label
                className="font-juliusSansOne text-xl py-2"
                htmlFor="email"
              >
                Email Address:
              </label>
              <input
                className="pr-20 py-2 md:pr-40 md:my-2 md:col-span-2"
                name="email"
                onChange={[Function]}
                value=""
              />
              <label
                className="font-juliusSansOne text-xl py-2"
                htmlFor="comment"
              >
                Comment:
              </label>
              <textarea
                className="pr-20 py-2 md:pr-40 md:my-2 md:col-span-2"
                name="comment"
                onChange={[Function]}
                value=""
              />
              <div
                className="hidden md:block"
              />
              <div
                className="hidden md:block"
              />
              <div
                className="font-juliusSansOne"
              >
                <button
                  className="float-right py-2 px-2 mx-4 mt-4 rounded-lg bg-gray-300 hover:bg-gray-700 hover:text-white"
                  onClick={[Function]}
                  type="submit"
                >
                  Submit
                </button>
                <button
                  className="float-right py-2 px-2 mx-4 mt-4 rounded-lg bg-gray-300 hover:bg-gray-700 hover:text-white"
                  onClick={[Function]}
                >
                  Reset
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </section>
</div>
`;
